pkg/ocmmanager/ocm_upgrade_config_manager.go

```diff
@@ -0,0 +1,439 @@
+// UpgradeConfig against the Cluster Service's Upgrade Policy.
+// Returns an indication of if the upgrade config had changed during the refresh,
+// and indication of error if one occurs.
+func (s *osdUpgradeConfigManager) RefreshUpgradeConfig() (bool, error) {
+
+	cluster, err := getClusterFromOCMApi(s.client, s.httpClient, s.config)
+	if err != nil {
+		log.Error(err, "cannot obtain internal cluster ID")
+		return false, err
+	}
+	if cluster.Id == "" {
+		return false, fmt.Errorf("cluster ID not found via OCM")
+	}
+	if cluster.Version.ChannelGroup == "" {
+		return false, fmt.Errorf("cluster channel group not set or found via OCM")
+	}
+
+	// Retrieve the cluster's upgrade policy from Cluster Services
+	upgradePolicies, err := getClusterUpgradePolicies(cluster, s.httpClient, s.config)
+	if err != nil {
+		log.Error(err, "cannot fetch next upgrade status")
+		return false, err
+	}
+
+	// Apply the Upgrade Policies to the cluster
+	changed, err := applyUpgradePolicies(upgradePolicies, cluster, s.client)
+	if err != nil {
+		log.Error(err, "cannot apply upgrade config")
+		return false, err
+	}
+
+	return changed, nil
+}
+
+// Queries and returns the Upgrade Policy from Cluster Services
+func getClusterUpgradePolicies(cluster *clusterInfo, client *http.Client, cfg *ocmUpgradeConfigManagerConfig) (*upgradePolicyList, error) {
+
+	url, err := cfg.GetOCMBaseURL()
+	if err != nil {
+		return nil, fmt.Errorf("can't read OCM API url: %v", err)
+	}
+	url.Path = path.Join(url.Path, CLUSTERS_V1_PATH, cluster.Id, UPGRADEPOLICIES_V1_PATH)
+
+	request := &http.Request{
+		Method: "GET",
+		URL:    url,
+	}
+	response, err := client.Do(request)
+	if err != nil {
+		return nil, fmt.Errorf("can't query upgrade service: %v", err)
+	}
+	operationId := response.Header[OPERATION_ID_HEADER]
+
+	if response.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("received error code '%v' from OCM upgrade policy service, operation id '%v'", response.StatusCode, operationId)
+	}
+
+	if response.Body != nil {
+		defer response.Body.Close()
+	}
+
+	var upgradeResponse upgradePolicyList
+	decoder := json.NewDecoder(response.Body)
+	decoder.DisallowUnknownFields()
+	err = decoder.Decode(&upgradeResponse)
+	if err != nil {
+		log.Error(err, "unable to decode OCM upgrade policy response, operation id '%v'", operationId)
+		return nil, err
+	}
+
+	return &upgradeResponse, nil
+}
+
+// Applies the supplied Upgrade Policy to the cluster in the form of an UpgradeConfig
+// Returns an indication of if the policy being applied differs to the existing UpgradeConfig,
+// and indication of error if one occurs.
+func applyUpgradePolicies(upgradePolicies *upgradePolicyList, cluster *clusterInfo, c client.Client) (bool, error) {
+	// Fetch the current UpgradeConfig instance, if it exists
+	upgradeConfigs := &upgradev1alpha1.UpgradeConfigList{}
+	err := c.List(context.TODO(), upgradeConfigs, &client.ListOptions{})
```

Comment:
Can we move to using `UPGRADECONFIG_CR_NAME` to get the UpgradeConfig?

Reply:
changed in 281806f to only handle a single UpgradeConfig and will search for it with UPGRADECONFIG_CR_NAME